### PR TITLE
For #19554: Add SyncEngine.CreditCards to  sync supportedEngines.

### LINKED ISSUE #19554: [Bug] "Sync cards" is disabled when it's enabled for the account
STR:

* Have Fenix signed in to an existing account where credit-cards are configured to sync (ie, presumably configured that way on a desktop profile)
* Jump through the hoops to enable CC autofill on Fenix.
* View "Credit cards" preferences - note the "Sync cards" option - it should be enabled but is disabled.

(I'll try and dig into this, but opening it now so I don't forget. For all I know it is permanently disabled and there's a TODO to enable it :) Note that the state of this checkbox will not come from the new autofill component, but instead via the "sync manager" component)

### PR DESCRIPTION
**Pull request by mcarare**
Fixes #19554 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by mcarare** [comment]
@grigoryk `WorkManagerSyncManager`'s `doSync` throws "CreditCards store must be configured with a KeyProvider".
Is there any work to be done before we could implement this, or am I missing something here? Thank you!

**Comment by gabrielluong** [comment]
> @grigoryk `WorkManagerSyncManager`'s `doSync` throws "CreditCards store must be configured with a KeyProvider".
> Is there any work to be done before we could implement this, or am I missing something here? Thank you!

@mcarare Please see https://github.com/mozilla-mobile/android-components/commit/ccb53c768487953a9d4b4afefb3f7a7ad518db90#diff-36276d5dee1aafaeafd259a863cfc14c004982efd14120d26b4ece59d3f4719c

File: app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt [hunk 1]
```diff
@@ -91,16 +93,17 @@ class BackgroundServices(
...
 
     init {
-        // Make the "history", "bookmark", "passwords", "tabs" stores
+        // Make the "history", "bookmark", "passwords", "tabs", "credit cards" stores
         // accessible to workers spawned by the sync manager.
         GlobalSyncableStoreProvider.configureStore(SyncEngine.History to historyStorage)
         GlobalSyncableStoreProvider.configureStore(SyncEngine.Bookmarks to bookmarkStorage)
         GlobalSyncableStoreProvider.configureStore(SyncEngine.Passwords to passwordsStorage)
         GlobalSyncableStoreProvider.configureStore(SyncEngine.Tabs to remoteTabsStorage)
+        GlobalSyncableStoreProvider.configureStore(SyncEngine.CreditCards to creditCardsStorage)
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt:106]: https://github.com/mozilla-mobile/android-components/commit/ccb53c768487953a9d4b4afefb3f7a7ad518db90#diff-36276d5dee1aafaeafd259a863cfc14c004982efd14120d26b4ece59d3f4719c

```
GlobalSyncableStoreProvider.configureStore(            
   storePair = SyncEngine.CreditCards to creditCardsAddressesStorage,            
   keyProvider = lazy { creditCardKeyProvider }        
)
```

**Comment by mcarare** [comment]
Tested the change and after login, the setting is reflected in both the account settings screen and credit card settings screen.

**Review by gabrielluong — APPROVED** [review]

**Review by mhammond — APPROVED** [review]
looks fine to me, but I suspect only Grisha knows the dark corners well enough.

**Comment by mcarare** [comment]
I am going to merge this and let QA test the issue. If something else comes up we will also ping @grigoryk .

**Pull request merged**